### ISSUE TITLE
browserAction undefined if browser_action missing in manifest.json

### ISSUE THREAD

**Issue by djbrown** [mdn/content#32627]
### MDN URL

[Toolbar button](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/user_interface/Toolbar_button)

### What specific section or headline is this issue about?

[Specifying the browser action](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/user_interface/Toolbar_button#specifying_the_browser_action)

### What information was incorrect, unhelpful, or incomplete?

Missing information about the `browser_action` key itself, only that all properties are optional. If I miss this key in `manifest.json` then `browser.browserAction` is undefined in `background.js`:
![browser_action](https://github.com/mdn/content/assets/11715448/624d2efa-869d-44bc-b820-fb3dd064c099)

### What did you expect to see?

It should be stated, that the `browserAction` API is only available of the `browser_api` key/object is declared in the `manifest.json`.

### Do you have any supporting links, references, or citations?

- AFAIK Firefox didn't throw that error a year ago
- [Manifest keys/browser_action](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/browser_action)
- [API/browserAction](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/browserAction)
- mildly relevant: #30025

### Do you have anything more you want to share?

might also be a problem with other APIs

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/mozilla/add-ons/webextensions/user_interface/toolbar_button`
* MDN URL: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/user_interface/Toolbar_button
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/mozilla/add-ons/webextensions/user_interface/toolbar_button/index.md
* Last commit: https://github.com/mdn/content/commit/8e7726da0782dcf7d1ca273177df60e3ff0893df
* Document last modified: 2023-11-17T15:34:57.000Z

</details>

**Comment by djbrown** [comment]
AFAIK Firefox didn't throw that error a year ago

**Comment by Rob--W** [comment]
> AFAIK Firefox didn't throw that error a year ago

The `browser_action` key has always been a requirement for the `browserAction` API.

It would be a good idea to double-check that the manifest key requirement is well-documented in the articles of all extension APIs. I went through the list of manifest keys at https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json#list_of_manifest.json_keys and from my recollection, these are the keys with an API:

- action key / action API
- browser_action key / browserAction API
- commands key / commands API
- (declarative_net_request key NOT required for declarativeNetRequest API)
- devtools_page key / devtools API (devtools APIs are only available to devtools pages, this seems already documented in the articles)
- omnibox key / omnibox API
- page_action key / pageAction API
- sidebar_action key / sidebarAction API
- (storage key NOT required for storage API)
- (theme key NOT required for theme API, in fact `theme` key is mutually exclusive with being an extension)
- (user_scripts key is required for userScripts API, but not for the new one, covered by https://github.com/mdn/content/pull/31813)
- (permissions key has the same name as the`permissions` API, but the latter is available without manifest keys, even though it is connected to the host_permissions (MV3), optional_permissions, optional_host_permissions (MV3) keys)

**Comment by rebloor** [comment]
I have checked that list and aside from the toolbar and address bar button the requirement, or otherwise, for the manifest key to be specified before the API can be used are clearly documented. It appears that the requirements for these buttons were correctly and fully specified in the [UI descriptions](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/user_interface). I have now submitted a PR to clarify this in the APIand manifest key descriptions.
